src/journal.c

```diff
@@ -293,264 +292,267 @@ static void JournalEntry_SaveTrainer(JournalEntry *journalEntry, JournalEntryTra
+    u32 *locationEvent = JournalEntry_GetEmptyLocationEvent(locationEvents);
+    *locationEvent = (journalEntryLocationEvent->locationID << 16) | ((journalEntryLocationEvent->trainerID & TRAINER_ID) << 6) | (journalEntryLocationEvent->eventType & EVENT_TYPE);
 }
 
-static void sub_0202B9F0(u32 *param0, UnkStruct_0202BCC8 *data)
+static void JournalEntry_SaveLocationEventEliteFour(u32 *locationEvents, JournalEntryLocationEvent *journalEntryLocationEvent)
 {
-    u32 *v0 = NULL;
+    u32 *locationEvent = NULL;
     u16 i;
 
-    for (i = 0; i < 4; i++) {
-        if ((param0[i] & 0x3f) == 11) {
-            param0[i] = 0;
+    for (i = 0; i < MAX_JOURNAL_LOCATION_EVENTS; i++) {
+        if ((locationEvents[i] & EVENT_TYPE) == LOCATION_EVENT_BEAT_ELITE_FOUR_MEMBER) {
+            locationEvents[i] = 0;
 
-            for (i = i; i < 4 - 1; i++) {
-                param0[i] = param0[i + 1];
-                param0[i + 1] = 0;
+            for (i = i; i < MAX_JOURNAL_LOCATION_EVENTS - 1; i++) {
+                locationEvents[i] = locationEvents[i + 1];
+                locationEvents[i + 1] = 0;
             }
 
             break;
         }
     }
 
-    v0 = sub_0202B91C(param0);
-    *v0 = (data->unk_02 << 16) | ((data->unk_04 & 0x3ff) << 6) | (data->unk_00 & 0x3f);
+    locationEvent = JournalEntry_GetEmptyLocationEvent(locationEvents);
+    *locationEvent = (journalEntryLocationEvent->locationID << 16) | ((journalEntryLocationEvent->trainerID & TRAINER_ID) << 6) | (journalEntryLocationEvent->eventType & EVENT_TYPE);
 }
 
-static void sub_0202BA48(u32 *param0, UnkStruct_0202BCC8 *data)
+static void JournalEntry_SaveLocationEventGym(u32 *locationEvents, JournalEntryLocationEvent *journalEntryLocationEvent)
 {
-    u32 *v0 = NULL;
+    u32 *locationEvent = NULL;
     u16 i;
 
-    for (i = 0; i < 4; i++) {
-        if (((param0[i] & 0x3f) == 9) && (((param0[i] >> 16) & 0xffff) == data->unk_02)) {
-            param0[i] = 0;
+    for (i = 0; i < MAX_JOURNAL_LOCATION_EVENTS; i++) {
+        if ((locationEvents[i] & EVENT_TYPE) == LOCATION_EVENT_GYM_WAS_TOO_TOUGH
+            && ((locationEvents[i] >> 16) & LOCATION_ID) == journalEntryLocationEvent->locationID) {
+            locationEvents[i] = 0;
 
-            for (i = i; i < 4 - 1; i++) {
-                param0[i] = param0[i + 1];
-                param0[i + 1] = 0;
+            for (i = i; i < MAX_JOURNAL_LOCATION_EVENTS - 1; i++) {
+                locationEvents[i] = locationEvents[i + 1];
+                locationEvents[i + 1] = 0;
             }
             break;
         }
     }
 
-    v0 = sub_0202B91C(param0);
-    *v0 = (data->unk_02 << 16) | ((data->unk_04 & 0x3ff) << 6) | (data->unk_00 & 0x3f);
+    locationEvent = JournalEntry_GetEmptyLocationEvent(locationEvents);
+    *locationEvent = (journalEntryLocationEvent->locationID << 16) | ((journalEntryLocationEvent->trainerID & TRAINER_ID) << 6) | (journalEntryLocationEvent->eventType & EVENT_TYPE);
```

Comment:
Define and use a macro to construct a location event, if appropriate.

Reply:
Sorry what do you mean here exactly? I'm not sure how a macro would be used here. From what I've seen so far, macros are for example the `ScrCmd`s that can be used in script, but I could definitely be missing something.

Reply:
I had something like this in mind (untested)
```c
#define LOCATION_EVENT(locationID, trainerID, eventType) (((locationID) << 16) | (((trainerID) & 0x3ff) << 6) | ((eventType) & 0x3f))
```

Reply:
Okay I've added some macros, let me know if there's anything I should change about them.